src/Panel/TimeLayerSliderPanel/TimeLayerSliderPanel.tsx

```diff
@@ -0,0 +1,496 @@
+
+import {
+  Select,
+  DatePicker,
+  Popover
+} from 'antd';
+const RangePicker = DatePicker.RangePicker;
+const Option = Select.Option;
+
+import './TimeLayerSliderPanel.less';
+
+type timeRange = [moment.Moment, moment.Moment];
+
+export interface Tooltips {
+  hours: string;
+  days: string;
+  weeks: string;
+  months: string;
+  years: string;
+  setToNow: string;
+  dataRange: string;
+}
+
+export interface DefaultTimeLayerSliderPanelProps {
+  className: string;
+  onChange: (arg: moment.Moment) => void;
+  timeAwareLayers: any[];
+  value: moment.Moment;
+  dateFormat: string;
+  tooltips: Tooltips;
+}
+
+export interface TimeLayerSliderPanelProps extends Partial<DefaultTimeLayerSliderPanelProps> {
+  map: any;
+  initStartDate: moment.Moment;
+  initEndDate: moment.Moment;
+}
+
+export interface TimeLayerSliderPanelState {
+  value: moment.Moment;
+  playbackSpeed: string;
+  autoPlayActive: boolean;
+  startDate: moment.Moment;
+  endDate: moment.Moment;
+}
+
+/**
+ * mapStateToProps - mapping state to props of Map Component.
+ *
+ * @param {Object} state current state
+ * @return {Object} mapped props
+ */
+
+/**
+ * The panel combining all time slider related parts.
+ */
+export class TimeLayerSliderPanel extends React.Component<TimeLayerSliderPanelProps, TimeLayerSliderPanelState> {
+
+  private _TimeLayerAwareSlider: any;
+  private _wmsTimeLayers: any[];
+  private _interval: number;
+
+  /**
+   * The default props of LayerSetBaseMapChooser
+   *
+   * @static
+   * @type {DefaultLayerSetBaseMapChooserProps}
+   * @memberof LayerSetBaseMapChooser
+   */
+  public static defaultProps: DefaultTimeLayerSliderPanelProps = {
+    className: '',
+    onChange: () => {/* empty */},
+    timeAwareLayers: [],
+    value: moment(moment.now()),
+    dateFormat: 'YYYY-MM-DD',
+    tooltips: {
+      setToNow: 'Set to now',
+      hours: 'Hours',
+      days: 'Days',
+      weeks: 'Weeks',
+      months: 'Months',
+      years: 'Years',
+      dataRange: 'Set data range'
+    }
+  };
+
+  /**
+   * Constructs time panel.
+   */
+  constructor(props: TimeLayerSliderPanelProps) {
+    super(props);
+
+    this.state = {
+      value: moment().milliseconds(0),
+      playbackSpeed: '1',
+      autoPlayActive: false,
+      startDate: moment().milliseconds(0),
+      endDate: moment().milliseconds(0),
+    };
+    this._interval = 1000;
+
+    this.wrapTimeSlider();
+
+    // binds
+    this.onTimeChanged = this.onTimeChanged.bind(this);
+    this.autoPlay = this.autoPlay.bind(this);
+    this.updateDataRange = this.updateDataRange.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      initStartDate,
+      initEndDate
+    } = this.props;
+
+    this.setState({
+      startDate: initStartDate,
+      endDate: initEndDate
+    });
+  }
+
+  componentDidUpdate(prevProps: TimeLayerSliderPanelProps) {
+    // TODO: this deep check may impact performance..
+    if (!(_isEqual(prevProps.timeAwareLayers, this.props.timeAwareLayers))) {
```

Comment:
You may want to compare the `ol_uid` of each layer instead.

Reply:
Comparing the `ol_uid` of each layer in `timeAwareLayers` of `prevProps` against the layer at the same position of `this.props`